tests/test_security_path_traversal.py

```
@@ -204,3 +204,64 @@ def test_run_tailer_path_validation(tmp_path):
     with pytest.raises(ValueError, match="run_id"):
         tailer.tail_run("..")
+
+def test_evolution_api_path_traversal_integration():
+    """Test evolution API against path traversal."""
+    import asyncio
+    from fastapi import HTTPException
+    from swarm.api.routes import evolution
+    from unittest.mock import patch, MagicMock
+    from pathlib import Path
+
+    with patch("swarm.api.routes.evolution._get_runs_root", return_value=Path("/tmp/runs")), \
```

Ruff (0.15.4)

[error] Probable insecure usage of temporary file or directory: "/tmp/runs"

(S108)